File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& n, int target) {
        int s=n.size();
        int l=0;
        int r=s-1;
        sort(n.begin(),n.end());
        while(l<r){
            int sum=n[l]+n[r];
            if(sum==target){
                return {l+1,r+1};
            }
            else if(sum>target){
                r--;
            }else{
                 l++;
            }
        }
        return {-1,-1};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] n, int target) {
        int l=0;
        int r=n.length-1;
        while(l<r){
            int sum=n[l]+n[r];
            if(sum==target){
                return new int[]{l+1,r+1};
            }else if(sum<target){
                l++;
            }
            else{
                r--;
            }
        }
        return new int[]{-1,-1};
    }
}